loadsave.c

```diff
@@ -50,8 +54,23 @@ create_directory_listing(uint8_t *data)
 	*data++ = 'P';
 	*data++ = 'C';
 	*data++ = 0;
+    
+    char filename[256];
+    uint8_t len = MIN(RAM[FNLEN], sizeof(filename) - 1);
+    memcpy(filename, (char *)&RAM[RAM[FNADR] | RAM[FNADR + 1] << 8], len);
+    filename[len] = 0;
+    
+    //use correct file location for ios
+    #if __APPLE__ && (TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE)
+    strcpy(filename,getenv("HOME"));
+    //concatenating the path string returned from HOME
+    strcat(filename,"/Documents/.");
+    #else
+    strcpy(filename,".");
```

Comment:
So I'd rather declare filename as a `char filename[256]` in the iOS path, and strcpy/strcat into it, and declare it as `const char *filename = "."` in the default path. This way, we don't allocate 256 bytes on the stack just to fill two bytes with constant data at runtime.

Reply:
👍